scripts/quests/otherAreas/RQ2_Way_of_the_Cook.lua

```diff
@@ -0,0 +1,119 @@
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Rycharde'] =
+            {
+                onTrigger = function(player, npc)
+                    if player:getCharVar("Quest[4][0]DayCompleted") + 7 < VanadielUniqueDay() then
+                        return quest:event(76, xi.items.DHALMEL_MEAT, xi.items.BEEHIVE_CHIP) -- Way of the Cook starting event.
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [76] = function(player, csid, option, npc)
+                    if option == 74 then -- Accept quest option.
+                        player:setCharVar("Quest[4][0]DayCompleted", 0)         -- Delete previous quest (Rycharde the Chef) variables
+                        quest:setVar(player, 'HourStarted', VanadielHour())        -- Set current quest started variables
+                        quest:setVar(player, 'DayStarted', VanadielDayOfTheYear()) -- Set current quest started variables
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    -- Section: Quest accepted. Handle trade and time limit
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+            ['Rycharde'] =
+            {
+                onTrigger = function(player, npc)
+                        daysPassed     = VanadielDayOfTheYear() - quest:getVar(player, "DayStarted")
+                        totalHoursLeft = 72 - (VanadielHour() + daysPassed * 24) + quest:getVar(player, "HourStarted")
+
+                        if totalHoursLeft > 0 then
+                            return quest:event(78, totalHoursLeft) -- You have x hours left.
+                        else
+                            return quest:event(79) -- Not done yet.
+                        end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, {xi.items.DHALMEL_MEAT, xi.items.BEEHIVE_CHIP}) then
+                        daysPassed     = VanadielDayOfTheYear() - quest:getVar(player, "DayStarted")
+                        totalHoursLeft = 72 - (VanadielHour() + daysPassed * 24) + quest:getVar(player, "HourStarted")
+
+                        if totalHoursLeft > 0 then
+                            return quest:event(80) -- Quest completed in time.
+                        else
+                            return quest:event(81) -- Quest completed late.
+                        end
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [80] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:tradeComplete()
+                        player:addGil(GIL_RATE*1500)
```

Comment:
The quest.rewards can handle the gil reward and the messaging

Reply:
The quest Gil reward varies depending on the time it took to complete. I was under the impresion that quest reward does not support the use of conditionals inside it